tests/c/setlongjmp.newcg.c

```diff
@@ -7,7 +7,7 @@
 //     jitstate: start-tracing
 //     we jumped
 //     jitstate: stop-tracing
-//     jitstate: trace-compilation-aborted: longjmp encountered
```

Comment:
I'm not sure about this. The jit-state prints (when enabled) are incomplete with this change.

I'm also not certain what the best cause of action is though.

Is `ykd` expensive at runtime?

Reply:
Perhaps we should split the jit-state prints into two categories (a log level, if you will):
 - those only printed when `ykd` is on.
 - those you always want to print (like this one).

But I think the messages should be formatted the same for both, with a `jit-state: ` prefix.

Reply:
Sorry, we raced.

OK, so how about this, we print the error as you have, but also emit a `jit-state` line as before. That way the `jit-state` info is complete, and someone grepping for `jit-state` won't be confused?

Reply:
I disagree: these fundamentally aren't "jit-state" things IMHO and shouldn't be presented as such. The JIT hasn't changed state as a result of these errors: the JIT state changed earlier (from tracing -> not tracing etc); subsequently an error has happened.

FWIW, I agree that there are probably more kinds of error/warning that need to be thought of than we currently have. For example "trace too long" is kind-of an expected thing and not something we want to print out all the time. But currently I have very little sense of how often it happens as I rarely have JIT state logging on.

Reply:
I agree that it would still be useful to know that an error has lead the jit to abort a trace. `yk warning: longjmp encountered` doesn't really tell us that this lead to the trace being aborted. But that might be crucial information for an end-user to figure out why their program isn't faster.

Reply:
Though I also agree that this technically isn't a jitstate. How about we write something like:
`yk warning: longjmp encountered (trace aborted)`.